app/controllers/claims_controller.rb

```diff
@@ -113,4 +117,18 @@ def claim_slug_sequence
   def prepend_view_path_for_policy
     prepend_view_path("app/views/#{current_policy_routing_name.underscore}")
   end
+
+  def generate_one_time_password
+    if params[:slug] == "email-address" && current_claim.has_ecp_policy?
+      one_time_password = generate_otp
+      ClaimMailer.ecp_email_verification(current_claim, one_time_password).deliver_now
+      session[:sent_one_time_password_at] = Time.now
+    end
+  end
+
+  def store_in_session_one_time_password
```

Comment:
There's obviously a lot of model logic in this controller but, that aside, I thought we concluded there's no need to store generated OTPs. Would be good to know why we need to now?

Reply:
@nomasprime We need to store the OTP so we can determine if the reason we have nil is because it has timed out vs its a wrong challenge code